__tests__/OmnichannelChatSDK.spec.ts

```diff
@@ -3296,5 +3296,162 @@ describe('Omnichannel Chat SDK', () => {
+
+            try {
+                await chatSDK.getAgentAvailability();
+                throw("Should throw error.");
+            } catch (ex) {
+                expect(ex?.message).toEqual("GetAgentAvailability is supported only for authenticated live chat widget.")
+            }
+        });
+
+        it('ChatSDK.getAgentAvailability() should throw error if auth token not set', async () => {
+            const chatSDKConfig = {
+                telemetry: {
+                    disable: true
+                },
+                chatReconnect: {
+                    disable: false,
+                },
+                getAuthToken: async () => {
+                    return 'authenticatedUserToken'
+                }
+            };
+
+            const dummyConfig = {
+                ChatWidgetLanguage: {
+                    msdyn_localeid: "1033"
+                }
+            };
+
+            const chatSDK = new OmnichannelChatSDK(omnichannelConfig, chatSDKConfig);
+            chatSDK.getChatConfig = jest.fn();
+
+            await chatSDK.initialize();
+            chatSDK.authSettings = {};
+
+            expect(chatSDK.authenticatedUserToken).toBe(null);
+            jest.spyOn(console, 'error');
+
+            chatSDK.liveChatConfig = dummyConfig;
+
+            try {
+                await chatSDK.getAgentAvailability();
+                throw("Should throw error.");
+            } catch (ex) {
+                expect(ex?.message).toEqual("Missing AuthToken for GetAgentAvailability.")
+            }
+        });
+
+        it('ChatSDK.getAgentAvailability() should throw error if conversation already started', async () => {
+            const chatSDKConfig = {
+                telemetry: {
+                    disable: true
+                },
+                chatReconnect: {
+                    disable: false,
+                },
+                getAuthToken: async () => {
+                    return 'authenticatedUserToken'
+                }
+            };
+
+            const dummyConfig = {
+                ChatWidgetLanguage: {
+                    msdyn_localeid: "1033"
+                }
+            };
+
+            const chatSDK = new OmnichannelChatSDK(omnichannelConfig, chatSDKConfig);
+            chatSDK.getChatConfig = jest.fn();
+
+            await chatSDK.initialize();
+            chatSDK.authSettings = {};
+            chatSDK.authenticatedUserToken = {};
+            chatSDK.conversation = {};
+            jest.spyOn(console, 'error');
+
+            chatSDK.liveChatConfig = dummyConfig;
+
+            try {
+                await chatSDK.getAgentAvailability();
+                throw("Should throw error.");
+            } catch (ex) {
+                expect(ex?.message).toEqual("GetAgentAvailability can only be called before a chat has started.")
+            }
+        });
+
+        it('ChatSDK.getAgentAvailability() should throw error if conversation already started', async () => {
```

Comment:
is the name correct?, is the same as LoC 3372 but different logic.

Reply:
fixed